lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.R

```diff
@@ -0,0 +1,1549 @@
+    
+    # the default area option
+    #slot.names <- names(gp);
+    #if (! ("col" %in% slot.names)) gp$col <- "grey";
+    
+    # TODO: adds grid objects as components
+    # creates a child gTree for major marks
+    gcm <- gTree(
+         name=nm, 
+         children = gList(),
+         childrenvp = vp,
+         gp = gp,
+         vp = NULL,
+         cl = NULL);
+
+    # creates a child gTree for minor marks
+    gcs <- gTree(
+         name=ns, 
+         children = gList(),
+         childrenvp = vp,
+         gp = gp,
+         vp = NULL,
+         cl = NULL);
+
+    # creates a child gTree for gridlines
+    gcl <- gTree(
+         name=nl, 
+         children = gList(),
+         childrenvp = vp,
+         gp = gp,
+         vp = NULL,
+         cl = NULL);
+    
+    # adds major and minor marks
+    cur <- min(scale);
+    if (breaks > 0)
+        width <- abs(scale[1] - scale[2]) / breaks
+    else
+        width <- abs(scale[1] - scale[2]);
+        
+    i = 0;
+    while(cur <= max(scale))
+    {
+        #print(cur); # for debug
+        
+        # create a major and minor mark
+        if (i == 0)
+        {
+            # as a major mark
+            if (right)
+            {
+                if(inner) m <- linesGrob(x=unit(c(1 - msize, 1), "npc"), y=unit(c(cur, cur), "native"), name=paste("t", cur, sep=""), gp=gp, vp=vp)
+                else m <- linesGrob(x=unit(c(1, 1 + msize), "npc"), y=unit(c(cur, cur), "native"), name=paste("t", cur, sep=""), gp=gp, vp=vp);
+                
+                gcm <- addGrob(gcm, m);
+            }
+            
+            if (left)
+            {
+                if(inner) m <- linesGrob(x=unit(c(0, msize), "npc"), y=unit(c(cur, cur), "native"), name=paste("b", cur, sep=""), gp=gp, vp=vp)
+                else m <- linesGrob(x=unit(c(-msize, 0), "npc"), y=unit(c(cur, cur), "native"), name=paste("b", cur, sep=""), gp=gp, vp=vp);
+                
+                gcm <- addGrob(gcm, m);
+            }
+        }
+        else
+        {
+            # as a minor mark
+            if (right)
+            {
+                if(inner) m <- linesGrob(x=unit(c(1 - ssize, 1), "npc"), y=unit(c(cur, cur), "native"), name=paste("t", cur, sep=""), gp=gp, vp=vp)
+                else m <- linesGrob(x=unit(c(1, 1 + ssize), "npc"), y=unit(c(cur, cur), "native"), name=paste("t", cur, sep=""), gp=gp, vp=vp);
+                
+                gcs <- addGrob(gcs, m);
+            }
+            
+            if (left)
+            {
+                if(inner) m <- linesGrob(x=unit(c(0, ssize), "npc"), y=unit(c(cur, cur), "native"), name=paste("b", cur, sep=""), gp=gp, vp=vp)
+                else m <- linesGrob(x=unit(c(-ssize, 0), "npc"), y=unit(c(cur, cur), "native"), name=paste("b", cur, sep=""), gp=gp, vp=vp);
+                
+                gcs <- addGrob(gcs, m);
+            }
+        }
+        
+        # creates a gridline
+        if (! is.null(lty))
+        {

```

Comment:
In general, the preferred TGI style is to do:

    if(statement){
        code
    }

with the open parentheses on the same line as the statement (makes for more compact code).  I'm not going to ask you to clean all of these up at this point, but it's something to note next time you write a script.